src/formTimingTracking.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Form timing tracking
+ * 
+ * This script analyses the interactions with <form> fields.
+ */
+
+ export default (subscribe, {formNameAttribute, fieldNameAttribute}) => {
+    let field_timings = {};
+    
+    const ppas_track_form_field_entry = function (e){
+        let fieldName = get_field_label(e.target);
+        field_timings[fieldName] = new Date().getTime();
+    };
+    
+    const ppas_track_form_field_leave = function(e){
+        let leaveType = e.type;
+        let formName = get_form_name(e.target);
+        let fieldName = get_field_label(e.target);
+        // eslint-disable-next-line no-prototype-builtins
+        if (field_timings.hasOwnProperty(fieldName)) {
+            let timeSpent = new Date().getTime() - field_timings[fieldName];
+          if (timeSpent > 0 && timeSpent < 1800000) {
+            subscribe({formName, fieldName, leaveType, timeSpent});
+          }
+          delete field_timings[fieldName];
+        }
+    };
+    
+    const get_field_label = function(field) {
```

Comment:
you can write it as one liner `const getFieldLabel = field => field.getAttribute(fieldNameAttribute)`